File: leetcode/Linked_List/0019_Remove_Nth_Node_From_End_of_List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        vector<ListNode*> list;
        ListNode *current = head;
        while (current != NULL) {
        	list.push_back(current);
        	current = current -> next;
        }
        if (n == list.size())
        	return head -> next;
        else if (n == 1)
        	list[list.size() - 2] -> next = NULL;
        else
        	list[list.size() - n - 1] -> next = list[list.size() - n - 1] -> next -> next;
		return head;    	
    }
};

File: leetcode/Linked_List/0019_Remove_Nth_Node_From_End_of_List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: Optional[ListNode]
        :type n: int
        :rtype: Optional[ListNode]
        """
        nodes = []
        current = head
        while current:
            nodes.append(current)
            current = current.next

        N = len(nodes)
        if n == N:
            return head.next
        elif n == 1:
            nodes[-2].next = None
            return head
        else:
            nodes[N - n - 1].next = nodes[N - n + 1]
            return head
